modules/_labs/core/react/lib/CanvasProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import {ThemeProvider} from 'emotion-theming';
+import {InputProvider} from '@workday/canvas-kit-react-core';
+import {CanvasTheme} from './theming/types';
+import {defaultCanvasTheme} from './theming/theme';
+
+export interface CanvasProviderProps {
+  theme: CanvasTheme;
+}
+
+export default class CanvasProvider extends React.Component<CanvasProviderProps> {
+  static defaultProps = {
+    theme: defaultCanvasTheme,
+  };
+
+  render() {
+    const {children, theme} = this.props;
+
+    return (
+      <ThemeProvider theme={theme}>
+        <InputProvider>{children}</InputProvider>
```

Comment:
Multiple instances of `InputProvider` will work fine from a consumer level, but I can't speak to the performance implications of doing that.

We could easily add a check in `InputProvider` to noop if one exists higher in the tree. This would be my vote rather than adding prop complexity to `CanvasProvider` or removing `CanvasProvider` completely.

I realize `CanvasProvider` is not doing much right now. However, there are two reasons I would like to keep it:
- Future proofing. We will generally always need some sort of top-level component. I assume in the future we will need to add more to it. When we do, it's more likely that consumers will get that future functionality for free. 
- Reduces the complexity from the consumers end. They don't really need to know about the theme and input provider if they're not using it.
```tsx
<CanvasProvider>
  // My App
</CanvasProvider>
```
is much simpler and hides our noise, compared to:
```tsx
<ThemeProvider>
  <InputProvider>
    <FutureStuff?>
      // My App
    </FutureStuff?>
  </InputProvider>
</ThemeProvider>
```

Anyway, let me know if you agree. I will probably add that functionality to `InputProvider` either way.

Reply:
@lychyi noop added to `InputProvider` in https://github.com/Workday/canvas-kit/pull/272/commits/6c2d52513309feb1fdcadff9548d814a6faa52e1